python/legate_dataframe/lib/core/column.pyx

```diff
@@ -106,6 +109,29 @@ cdef class LogicalColumn:
                 "from_cudf() only supports cudf columns and device scalars."
             )
 
+    @staticmethod
+    def from_arrow(array) -> LogicalColumn:
```

Comment:
```suggestion
    def from_arrow(pa.Array array not None) -> LogicalColumn:
```
Maybe?  Then Cython will give a nice error for the type.  (But it seems arrow handles bad types fine, so also OK.)